build.go

```diff
@@ -119,24 +96,22 @@ func Build(entryResolver EntryResolver,
 			logger.Process(fmt.Sprintf("Reusing cached layer %s", sdkLayer.Path))
 			logger.Break()
 
-			err = dotnetSymlinker.Link(context.WorkingDir, sdkLayer.Path)
+			err = fs.Copy(filepath.Join(sdkLayer.Path, "dotnet"), filepath.Join(context.WorkingDir, ".dotnet_root", "dotnet"))
 			if err != nil {
 				return packit.BuildResult{}, err
 			}
 
-			envLayer.SharedEnv.Prepend("PATH",
-				filepath.Join(context.WorkingDir, ".dotnet_root"),
-				string(os.PathListSeparator))
-
-			envLayer.SharedEnv.Override("DOTNET_ROOT", filepath.Join(context.WorkingDir, ".dotnet_root"))
+			sdkLayer.BuildEnv.Prepend("PATH", sdkLayer.Path, string(os.PathListSeparator))
+			logger.EnvironmentVariables(sdkLayer)
+			envLayer.LaunchEnv.Prepend("PATH", filepath.Join(context.WorkingDir, ".dotnet_root"), string(os.PathListSeparator))
```

Comment:
Is setting one in the BuildEnv and the other in the LaunchEnv to prevent a case where both layers put a version of the `dotnet` CLI on the `PATH`?

Reply:
Not quite, it's more to prevent more than what's needed being available during launch. The `dotnet` CLI in both cases is the same. This is confusing and I would've liked to find a way to avoid doing this. The reason for this code is:
1. The first case is in the .NET Publish case, we want the full SDK hive, so we make the entire SDK layer available on the `PATH` during build.
2. The less obvious case is in the .NET Execute case (FDD case specifically), the `dotnet` CLI is needed at launch-time, so we just copy it into the  `/workspace/.dotnet_root` to isolate it, and make it accessible via the `PATH`. This `envLayer` is ALWAYS marked for `launch`.

Reply:
If you can think of a nicer way to just have the `dotnet` CLI available at launch-time, let me know, I'm all ears.